### PR TITLE
Feature/mirror object interaction enhancements

### PR DESCRIPTION
**Pull request by FishError**
Refactored object interaction code to hopefully make future interaction easier. Changed moving object through mirror logic and not includes forward backwards movement with mouse wheel (xyz movement is based on player position relative to the mirror). Also implemented objects dropping  on collision if a player is moving it.

### CODE REVIEW

File: SelfReflection/SelfReflection/Assets/Scripts/Raycast/MoveObjectController.cs [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+    {
+        sensX = transform.GetComponent<PlayerCam>().sensX;
+        sensY = transform.GetComponent<PlayerCam>().sensY;
+    }
+
+    private void Update()
+    {
+        if (Input.GetMouseButtonDown(0))
+        {
+            //print("Left Click");
```
Comment (KaifengZhou) [inline SelfReflection/SelfReflection/Assets/Scripts/Raycast/MoveObjectController.cs:34]: Remove Prints

Comment (FishError) [reply SelfReflection/SelfReflection/Assets/Scripts/Raycast/MoveObjectController.cs:34]: those were jacky's prints so idk if he wants them removed or not

Comment (KaifengZhou) [reply SelfReflection/SelfReflection/Assets/Scripts/Raycast/MoveObjectController.cs:34]: he forgot to remove them last time, and we were short on time so we just approved it

Comment (FishError) [reply SelfReflection/SelfReflection/Assets/Scripts/Raycast/MoveObjectController.cs]: removed

**Review by xcrowy — APPROVED** [review]

**Review by KaifengZhou — APPROVED** [review]

**Pull request merged**